pkg/kp/rcstore/store.go

```diff
@@ -21,6 +21,19 @@ type Store interface {
 	Get(id fields.ID) (fields.RC, error)
 	List() ([]fields.RC, error)
 
+	// Watch for changes to the entire store and output the current list of RCs
+	// any time it changes. This function blocks; close the quit channel to
+	// terminate. Callers must drain the returned channels, including the error
+	// channel.
+	WatchNew(quit <-chan struct{}) (<-chan []fields.RC, <-chan error)
+
+	// Use the given session string (which uniquely identifies the lock holder)
+	// to acquire a lock on the specified replication controller. If the lock
+	// was successfully claimed, return (true, nil). If the request completed
+	// but the lock was already held, return (false, nil). Otherwise return an
+	// appropriate error.
+	Lock(id fields.ID, session string) (bool, error)
```

Comment:
Are there any guidelines that callers should follow regarding the session used to acquire the lock?  Any guidelines for what happens if the session is terminated?